### PR TITLE
 Ahmed -week2-js

### PR DESCRIPTION
**Pull request by ahmedahashim**

### CODE REVIEW

File: week-2-practise/level-2/1-boolean.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // The code is valid but the program does not produce the expected result
-// Why doesn't it work?
+// Why doesn't it work? the value assigned to the variable isHappy is string it must be aboolean we removed "".
```
Comment (Atayo) [inline week-2-practise/level-2/1-boolean.js]: Well done, great work.

File: week-2/D-predicates/exercise.js [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
 // Finish the predicate function to test if the passed number is negative (less than zero)
 function isNegative(number) {
-
+  return typeof number > 0;
 }
 
 // Finish the predicate function to test if the passed number is between 0 and 10
 function isBetweenZeroAnd10(number) {
-
+  return typeof number > 0 < 10;
```
Comment (martynamaron) [inline week-2/D-predicates/exercise.js]: that's not correct - what happens when number is `-20`? And what happens when the number is `40`? Does it give you sensible answers? The clue is in `typeof` - why did you use it?

File: week-2/F-logical-operators/exercise2.js [hunk 3]
```diff
@@ -5,17 +5,42 @@
   Update the code so that you get the expected result.
 */
 
-function isNegative() {}
+function isNegative(num) {
+  if (num < 0) {
```
Comment (martynamaron) [inline week-2/F-logical-operators/exercise2.js]: Nice! how can you make your code shorter though? Remember that just `num < 0` returns a boolean and that's what you want this function to return too!

File: week-2/F-logical-operators/exercise2.js [hunk 4]
```diff
@@ -5,17 +5,42 @@
...
+
+function isLongName(name) {
+  if (name.lenght > 5);
+  {
+    return true;
+  }
+}
+var theName = "Daniel";
+function startsWithD(theName) {
+  if (theName.includes("D")) {
```
Comment (martynamaron) [inline week-2/F-logical-operators/exercise2.js]: The name of your function doesn't match what it actually does. Your function checks if the name **includes** the letter "D", not if the name **starts with it** - those are different things :)

File: week-2/F-logical-operators/exercise2.js [hunk 5]
```diff
@@ -5,17 +5,42 @@
...
-console.log("Is -10 is a negative number?", isNegative(-10));
-console.log("Is 5 a negative number?", isNegative(5));
-console.log("Is 10 in the range 5-10?" isBetween5and10(10))
-console.log("Is Daniel a short name?" isLongName("Daniel"));
-console.log("Does Daniel start with 'D'?" startsWithD("Daniel"));
+console.log("Is -10 is a negative number?" + isNegative(-10));
+console.log("Is 5 a negative number?" + isNegative(5));
+console.log("Is 10 in the range 5-10?" + isBetween5and10(10));
+console.log("Is Daniel a short name?" + isLongName("Daniel"));
+console.log("Does Daniel start with 'D'?" + startsWithD("Daniel"));
```
Comment (martynamaron) [inline week-2/F-logical-operators/exercise2.js]: Cool! `+` works too and nice spot! :)

File: week-2/G-conditionals-2/exercise-3.js [hunk 6]
```diff
@@ -9,7 +9,18 @@
 */
 
 function calculateGrade(mark) {
-
+  if (mark >= 80 && mark <= 100) {
```
Comment (martynamaron) [inline week-2/G-conditionals-2/exercise-3.js]: No need for ` mark <= 100` when you know that's the highest grade :) But it's not wrong!

File: week-2/G-conditionals-2/exercise-3.js [hunk 7]
```diff
@@ -9,7 +9,18 @@
...
+    return "A";
+  }
+  if (mark >= 60 && mark < 80) {
+    return "B";
+  }
+  if (mark >= 50 && mark < 60) {
+    return "C";
+  }
+  if (mark < 50) {
```
Comment (martynamaron) [inline week-2/G-conditionals-2/exercise-3.js]: Well done on paying attention to the `>=` and `<` signs!

File: week-2/I-array-properties/exercise.js [hunk 8]
```diff
@@ -6,7 +6,9 @@
 */
 
 function isEmpty(arr) {
-  return; // complete this statement
+  if (arr.length === 0) return true;
```
Comment (martynamaron) [inline week-2/I-array-properties/exercise.js]: Try and refactor this - `arr.length === 0)` already returns a boolean, so you don't really need the `if` statement ;)

File: week-2/J-array-get-set/exercise.js [hunk 9]
```diff
@@ -3,13 +3,14 @@
...
 
 function first(arr) {
-  return; // complete this statement
+  return arr[0]; // complete this statement
 }
 
-function last(arr) {
-  return; // complete this statement
+function last(names) {
+  return names[2]; // complete this statement
```
Comment (martynamaron) [inline week-2/J-array-get-set/exercise.js]: the `last` function is meant to return the **last** element in the array. What happens in this case?
```
var array = [1, 2, 3, 4, 5];
console.log(last(array));
```

Try again with the `last` function please :)

File: week-2/L-array-methods-2/exercise2.js [hunk 10]
```diff
@@ -7,7 +7,11 @@
   Tip: use the string method .split() and the array method .join()
 */
 
-function capitalise(str) {}
+function capitalise(str) {
+  str = str.split("");
+  str[0] = str[0].toUpperCase();
+  return str.join("");
```
Comment (martynamaron) [inline week-2/L-array-methods-2/exercise2.js]: Nice!

File: week-2/M-array-map/exercise.js [hunk 11]
```diff
@@ -10,7 +10,7 @@ function double(num) {
 }
 
 var numbers = [1, 2, 3, 4];
-var numbersDoubled; // complete this statement (use map and the double function)
+var numbersDoubled = numbers.map(double); // complete this statement (use map and the double function)
```
Comment (martynamaron) [inline week-2/M-array-map/exercise.js]: Well done!

File: week-2-practise/level-1/5-indexOf.js [hunk 12]
```diff
@@ -3,7 +3,7 @@
 
 var itemToFind = 7;
 var arr = [3, 5, 61, 7, 123];
-var index; // ONLY EDIT THIS LINE
+var index = arr.indexOf(7); // ONLY EDIT THIS LINE
```
Comment (martynamaron) [inline week-2-practise/level-1/5-indexOf.js]: Yes! 👏

File: week-2-practise/level-1/6-map.js [hunk 13]
```diff
@@ -11,7 +11,7 @@ function tidyUpString(str) {
 }
 
 var mentors = ["/Daniel ", "irina ", " Gordon", "ashleigh "];
-var mentorsTidy; // ONLY EDIT THIS LINE
+var mentorsTidy = mentors.map(tidyUpString); // ONLY EDIT THIS LINE
```
Comment (martynamaron) [inline week-2-practise/level-1/6-map.js]: Well done! Looks like you understand what `map()` does! :)

File: week-2-practise/level-2/6-map.js [hunk 14]
```diff
@@ -2,9 +2,11 @@
 // 1. Write a function that formats the numbers into a string with the percentage symbol appended e.g. "10%"
 // 2. Declare a new array, `percentagesFormatted`, containing
 //    - each item in `percentages` formatted by your function
-
+function addPercentage(num) {
+  return num + "%";
+}
 var percentages = [1, 23, 92, 18];
-
+var percentagesFormatted = percentages.map(addPercentage);
```
Comment (martynamaron) [inline week-2-practise/level-2/6-map.js]: Yay! Well done 👏

File: week-2-practise/level-3/2-validation.js [hunk 15]
```diff
@@ -6,7 +6,32 @@
...
+
+function isNumber(num) {
+  if (typeof num == "number") {
+    return true;
+  } else {
+    return false;
+  }
+}
+function isInteger(num) {
+  if (num % 1 === 0) {
```
Comment (martynamaron) [inline week-2-practise/level-3/2-validation.js]: Nice!

File: week-2-practise/level-3/2-validation.js [hunk 16]
```diff
@@ -6,7 +6,32 @@
...
+    return true;
+  } else {
+    return false;
+  }
+}
+function isNotInExcludedNums(num) {
+  if (excludedNums.includes(num)) {
```
Comment (martynamaron) [inline week-2-practise/level-3/2-validation.js]: Great! How can you reverse this logic using `!`?

File: week-2-practise/level-3/3-sort.js [hunk 17]
```diff
@@ -2,9 +2,12 @@
 // It should contain the values of `nums` and `nums2` in ascending order
 // Tip: you might need to read the documentation for .sort (search "mdn array sort")
 
+function myFunction(a, b) {
```
Comment (martynamaron) [inline week-2-practise/level-3/3-sort.js]: I think this function could do with a better name :)

File: week-2-practise/level-3/4-slice.js [hunk 18]
```diff
@@ -5,7 +5,7 @@
 // - insert the replacement value at the provided index
 
 function replace(arr, index, value) {
-  return; // complete this statement
+  return arr.slice(0, index).concat(value, arr.slice(index + 1)); // complete this statement
```
Comment (martynamaron) [inline week-2-practise/level-3/4-slice.js]: Well done :)

File: week-2-practise/level-3/5-indexOf.js [hunk 19]
```diff
@@ -4,19 +4,20 @@
 // - insert `newValue` at that index
 // - NOT change the original array
 // - return a new array with the replacement value inserted
-
 function replace(arr, valueToReplace, newValue) {
-  return; // complete this statement
+  var index = arr.indexOf(valueToReplace);
```
Comment (martynamaron) [inline week-2-practise/level-3/5-indexOf.js]: If the array doesn't contain the given value, it returns `-1`. And you can't use a negative index in an array - so how can you check for `-1` and print a nice error message to the terminal? :)
Extra bonus question!

File: week-2-practise/level-3/6-map.js [hunk 20]
```diff
@@ -12,10 +12,14 @@ function tidyUpString(str) {
...
 function captialise(str) {
   // complete this function
+  var myString = str.split("");
+  myString[0] = myString[0].toUpperCase();
+  return myString.join("");
 }
 
 var mentors = ["/Daniel ", "irina ", " Gordon", "ashleigh "];
-var mentorsTidyAndCapitalised;
+var theMentors = mentors.map(tidyUpString);
```
Comment (martynamaron) [inline week-2-practise/level-3/6-map.js]: This variable holds an array of strings that were tidied up - so what would be a better name for it?

File: week-2/F-logical-operators/exercise2.js (hunk 5 above)
Comment (khan-zareen) [reply week-2/F-logical-operators/exercise2.js]: Good use of logical operator

**Pull request closed without merging**